pkg/blockchain/blockchainPublisher_test.go

```diff
@@ -40,14 +48,15 @@ func buildPublisher(t *testing.T) (*blockchain.BlockchainPublisher, func()) {
 	require.NoError(t, err)
 
 	return publisher, func() {
+		defer cleanup()
 		cancel()
 		client.Close()
 	}
 }
 
 func TestPublishIdentityUpdate(t *testing.T) {
 	publisher, cleanup := buildPublisher(t)
-	defer cleanup()
+	t.Cleanup(cleanup)
```

Comment:
this is really cool!

Reply:
I don't know if that was always in the test library and I just didn't know it existed, or if it's new. But I agree.

We should really be using this everywhere, rather than returning cleanup functions